source/uwp/Renderer/lib/MediaHelpers.cpp

```diff
@@ -194,17 +194,55 @@ namespace AdaptiveCards::Rendering::Uwp::MediaHelpers
                     if (currentCaptionType == supportedCaptionTypes[i])
                     {
                         const auto timedTextURL = GetUrlFromString(renderContext.HostConfig(), captionSource.Url());
-                        const auto timedTextSrc = winrt::TimedTextSource::CreateFromUri(timedTextURL);
-                        timedTextSrc.Resolved(
-                            [captionSource](winrt::TimedTextSource const& /*sender*/,
-                                            winrt::TimedTextSourceResolveResultEventArgs const& args)
-                            {
-                                if (!args.Error())
+
+                        winrt::IAdaptiveCardResourceResolver resourceResolver{nullptr};
+                        if (const auto resourceResolvers = renderContext.ResourceResolvers())
+                        {
+                            resourceResolver = resourceResolvers.Get(timedTextURL.SchemeName());
+                        }
+
+                        if (resourceResolver == nullptr)
+                        {
+                            const auto timedTextSrc = winrt::TimedTextSource::CreateFromUri(timedTextURL);
+                            timedTextSrc.Resolved(
+                                [captionSource](winrt::TimedTextSource const& /*sender*/,
+                                                winrt::TimedTextSourceResolveResultEventArgs const& args)
+                                {
+                                    if (!args.Error())
+                                    {
+                                        args.Tracks().GetAt(0).Label(captionSource.Label());
+                                    }
+                                });
+                            mediaSrc.ExternalTimedTextSources().Append(timedTextSrc);
```

Comment:
We repeat this code twice: here and in case resolver is not null.
Could you please create a variable to hold this lambda before this logic and just pass it into `.Resolved()` instead?
I don't think it's worth extracting this as a method but lambda in a var will probably be fine :)

Reply:
Sounds good